### PR TITLE
driver: kineis_kim1: add kineis driver for KIM1 module

### PR DESCRIPTION
**Pull request by RomainPelletant**
Add satellite API to implement satellite sending strategy. Related to issue #52971
Signed-off-by: romain pelletant <romainp@kickmaker.net>

### CODE REVIEW

**Comment by RomainPelletant** [comment]
@carlescufi Connected too late on the meeting : could we postpone this topic to the next A. MG meeting?

File: include/zephyr/drivers/satellite.h [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+				   uint8_t number_of_send,
+				   k_timeout_t time_between_send,
+				   satellite_api_send_result_cb *result_cb);
+
+
+struct satellite_driver_api {
+	satellite_api_config config;
+	satellite_api_send_sync send_sync;
+	satellite_api_send_pool_async send_pool_async;
+};
```
Comment (fabiobaltieri) [inline include/zephyr/drivers/satellite.h:85]: Hey trying to anticipate a bit the architecture WG discussion: I think the question would be if this API makes sense for a generic class of satellite communication device. Don't think we have existing expertise in the community about this so it'd be cool to know: is this something that potentially map to other type of satellite modems?

I see you already split the device specific APIs out in `drivers/satellite/kim1.h` (good!) so it seems like that has been thought out already.

**Comment by RomainPelletant** [comment]
@MaureenHelm Just need to synchronize me 
When arch WG are scheduled please?

**Comment by MaureenHelm** [comment]
> @MaureenHelm Just need to synchronize me When arch WG are scheduled please?

https://github.com/zephyrproject-rtos/zephyr/wiki/Zephyr-Committee-and-Working-Group-Meetings

**Comment by carlescufi** [comment]
@RomainPelletant please let me know on Discord if you'd like this included, or drop me an email.

**Comment by MaureenHelm** [comment]
> @RomainPelletant please let me know on Discord if you'd like this included, or drop me an email.

@RomainPelletant Any update?

**Pull request closed without merging**